src/features/orcidlink/Home/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { orcidlinkAPI } from '../../../common/api/orcidlinkAPI';
+import { useAppSelector } from '../../../common/hooks';
+import { authUsername } from '../../auth/authSlice';
+import { usePageTitle } from '../../layout/layoutSlice';
+import ErrorMessage from '../common/ErrorMessage';
+import Home from './Home';
+
+export default function HomeController() {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const username = useAppSelector(authUsername)!;
+
+  usePageTitle('KBase ORCID Link');
+
+  const { data, error, isError, isSuccess } =
+    orcidlinkAPI.useOrcidlinkInitialStateQuery({ username });
```

Comment:
To skip the query when username is undefined
```suggestion
  import { skipToken } from '@reduxjs/toolkit/query';
  ...
  const { data, error, isError, isSuccess } =
    orcidlinkAPI.useOrcidlinkInitialStateQuery(username?{ username }:skipToken);
```

Reply:
Yes, but  username is never undefined! This component is only invoked when authenticated, so I thought it cleaner to make that assumption and simplify the code.

The code pathway ensures that the orcidlink feature is only ever mounted when the app is authenticated. Yet the app state overall includes being unauthenticated.

It simplifies the implementation to assume that the user is known. For instance, the orcidlink initial state does not make sense without the authenticated username. Even though the query could be skipped if username is undefined, the query would still have to handle the case of the username being undefined.

In other words, a bunch of bother if we let username be undefined, and it seems cleaner to just assert at the beginning that it will never be undefined.

Reply:
In that case maybe a `const usernameString = username ?? ''` would suffice? I'm very hesitant about assertions, as they can have really weird downstream effects, though probably not in this case. Just worried about precedent.

Reply:
How about:

```
  // look, no non-null assertion.
  const username = useAppSelector(authUsername);

  // this condition will never occur in real life; we can, however, test it.
  if (typeof username === 'undefined') {
    throw new Error('Impossible: username is not defined');
  }
```

I've used two approaches in this situation, other than a simple non-null assertion:

- if the value is whatever-is-impossible-but-TS-can't-know-this, throw an error with the word "impossible" somewhere in the error message; the exception will never be invoked in reality, and if it could be it would handled like any other uncaught runtime error by the error boundary; that is, the invalid value would not be propagated and do unexpected things.

- pass the auth state from the authentication wrapper component as a prop to the sub-component

Reply:
So I added the detection of undefined, throwing an exception, and a test to match the display of the message in the error boundary.